### PR TITLE
[compiler-rt] Adds builtins support for xros.

### PR DESCRIPTION
**Pull request by rohit-rao**
Adds support for xros when compiling builtins. This is disabled by default and controlled with COMPILER_RT_ENABLE_XROS, similar to watchOS/tvOS.

### CODE REVIEW

**Comment by rohit-rao** [comment]
@delcypher Would like to have support for xros, please take a look when you have a minute. Thanks!

**Comment by nico** [comment]
@jroelofs fyi

**Comment by nico** [comment]
@yln @kubamracek fyi too :)

**Comment by nico** [comment]
1. I think you want to edit `compiler-rt/cmake/base-config-ix.cmake` to add an `option()` for this, right after `option(COMPILER_RT_ENABLE_TVOS "Enable building for tvOS - Experimental" Off)`
2. You probably want to make a similar change to yours in `compiler-rt/cmake/config-ix.cmake` as well
3. (I'll note that you don't add a min ver flag like tvOS and watchOS do, but as far as I can tell, not even Xcode clang supports `-mros-version-min=` yet, so that'll have to wait until that's added and this is fine as-is.)

File: compiler-rt/cmake/builtin-config-ix.cmake [hunk 1]
```diff
@@ -162,6 +164,11 @@ if(APPLE)
       list(APPEND DARWIN_tvossim_BUILTIN_ALL_POSSIBLE_ARCHS arm64)
     endif()
   endif()
+  if(COMPILER_RT_ENABLE_XROS)
+    list(APPEND DARWIN_EMBEDDED_PLATFORMS xros)
+    set(DARWIN_xros_BUILTIN_ALL_POSSIBLE_ARCHS arm64)
```
Comment (dtellenbach) [inline compiler-rt/cmake/builtin-config-ix.cmake:169]: Please use `${ARM64} ${ARM32}` similar to what ios does.

File: compiler-rt/cmake/builtin-config-ix.cmake [hunk 2]
```diff
@@ -162,6 +164,11 @@ if(APPLE)
...
+    set(DARWIN_xrossim_BUILTIN_ALL_POSSIBLE_ARCHS ${X86_64} arm64)
```
Comment (nico) [inline compiler-rt/cmake/builtin-config-ix.cmake:170]: (Is there an xros simulator x86_64? I don't have an intel machine to test at hand – just curious.)

Comment (dtellenbach) [inline compiler-rt/cmake/builtin-config-ix.cmake:170]: Please use `arm64` only.

Comment (rohit-rao) [reply compiler-rt/cmake/builtin-config-ix.cmake]: Done.

File: compiler-rt/cmake/builtin-config-ix.cmake (hunk 1 above)
Comment (rohit-rao) [reply compiler-rt/cmake/builtin-config-ix.cmake]: Done.

File: compiler-rt/cmake/builtin-config-ix.cmake (hunk 2 above)
Comment (rohit-rao) [reply compiler-rt/cmake/builtin-config-ix.cmake]: Per comment below, no x86_64 (I think there was in early betas). Removed.

**Comment by ributzka** [comment]
> 3. (I'll note that you don't add a min ver flag like tvOS and watchOS do, but as far as I can tell, not even Xcode clang supports `-mros-version-min=` yet, so that'll have to wait until that's added and this is fine as-is.)

I don't think there ever will be a new `-m*-version-min=`, so there is no need to add a new option for every new platform.  The new recommended approach is to use `-target` (as used by the Xcode build system) or alternatively `-arch` in combination with `-mtargetos=`.

**Comment by ributzka** [comment]
See https://reviews.llvm.org/D106316

**Comment by nico** [comment]
This has been sitting for a while and got some comments that got addressed, so I'll merge this. Shout if you want follow-up work to happen :)

**Pull request merged**